test/testAdaptors/StakingAdaptors/EtherFiStakingAdaptor.t.sol

```diff
@@ -0,0 +1,352 @@
+        for (uint256 i; i < maxRequests; ++i) {
+            mintAmounts[i] = uint256(keccak256(abi.encodePacked(seed, i))) % 1_000e18; // Cap value to 1k.
+            expectedTotalAssets += mintAmounts[i];
+            deal(address(primitive), address(this), mintAmounts[i]);
+            cellar.deposit(mintAmounts[i], address(this));
+
+            _mintDeriviative(mintAmounts[i]);
+            burnAmount = derivative.balanceOf(address(cellar));
+            _startDeriviativeBurnRequest(burnAmount);
+
+            requests = etherFiAdaptor.getRequestIds(address(cellar));
+            assertEq(requests.length, i + 1, "Should have i + 1 requests.");
+        }
+
+        // Making 1 more burn request should revert.
+        _mintDeriviative(initialAssets);
+        burnAmount = derivative.balanceOf(address(cellar));
+        vm.expectRevert(bytes(abi.encodeWithSelector(StakingAdaptor.StakingAdaptor__MaximumRequestsExceeded.selector)));
+        _startDeriviativeBurnRequest(burnAmount);
+
+        // Finalize requests.
+        for (uint256 i; i < maxRequests; ++i) {
+            uint256 requestId = etherFiAdaptor.requestIds(address(cellar), i);
+
+            _finalizeRequest(requestId, mintAmounts[i]);
+        }
+
+        // Complete requests.
+        for (uint256 i; i < maxRequests; ++i) {
+            uint256 requestId = etherFiAdaptor.requestIds(address(cellar), 0);
+            _completeDerivativeBurnRequest(requestId);
+        }
+        requests = etherFiAdaptor.getRequestIds(address(cellar));
+        assertEq(requests.length, 0, "Should have no burn requests left.");
+        assertApproxEqRel(
+            cellar.totalAssets(),
+            expectedTotalAssets,
+            0.000001e18,
+            "totalAssets should not have changed."
+        );
+        uint256 expectedPrimitiveBalance = expectedTotalAssets - initialAssets;
+        uint256 expectedDerivativeBalance = initialAssets;
+        assertApproxEqRel(
+            primitive.balanceOf(address(cellar)),
+            expectedPrimitiveBalance,
+            0.000001e18,
+            "primitive balance should equal expected"
+        );
+        assertApproxEqRel(
+            derivative.balanceOf(address(cellar)),
+            expectedDerivativeBalance,
+            0.000001e18,
+            "derivative balance should equal expected"
+        );
+    }
+
+    function testWrappingAndUnwrapping(uint256 amount) external {
+        amount = bound(amount, 0.0001e18, 1_000e18);
+        deal(address(primitive), address(this), amount);
+        cellar.deposit(amount, address(this));
+
+        _mintDeriviative(amount);
+
+        uint256 derivativeBalance = derivative.balanceOf(address(cellar));
+
+        _wrap(type(uint256).max);
+
+        assertApproxEqAbs(derivative.balanceOf(address(cellar)), 0, 1, "All derivative should be wrapped.");
+        assertGt(wrappedDerivative.balanceOf(address(cellar)), 0, "Should have non zero wrapped derivative amount.");
```

Comment:
Why not an `assertApproxEqAbs ` test?

Reply:
I am more concerned with knowing that we did successfully wrap the derivative asset. Not so much that the underlying protocol did wrap at the exchange rate, since that would be a logical error in their code